frontend/javascripts/dashboard/dataset/dataset_import_view.js

```diff
@@ -162,10 +188,21 @@ class DatasetImportView extends React.PureComponent<Props, State> {
   }
 
   getDatasourceDiffAlert() {
+    // Only show if the option did not apply
+    const {
+      savedDataSourceOnServer,
+      differenceBetweenDatasources,
+      didApplyDifferenceBetweenDatasources,
+      inferredDataSource,
+    } = this.state;
+    const didUserAppliedSuggestions =
+      didApplyDifferenceBetweenDatasources === true &&
+      isDatasourceJSONValid(savedDataSourceOnServer);
```

Comment:
In which case would `didApplyDifferenceBetweenDatasources` be true but `isDatasourceJSONValid(savedDataSourceOnServer)` be false? In other words, I don't understand why the second condition is needed, yet.

Reply:
Oh thx you just found a bug in the logic. :bug: 

As I went over this again and not having worked with this PR for more than a week I came up the the idea of better handling the "state of the datasource"-setting. I just implemented this and this rework also affects the logic.
This means that you have to take a look at the whole logic again :see_no_evil: 
But hopefully this change makes the logic more expressive / simple and therefore easier to understand.

Please tell me what you think about this and if you have suggestions, that could improve the understandability even more :smile: